framework/include/systems/SolverSystem.h

```diff
@@ -39,6 +39,22 @@ class SolverSystem : public SystemBase
   virtual void stopSolve(const ExecFlagType & exec_flag,
                          const std::set<TagID> & vector_tags_to_close) = 0;
 
+  /**
+   * Returns the convergence state
+   * @return true if converged, otherwise false
+   */
+  virtual bool converged() = 0;
+
+  /**
+   * If the system has a kernel that corresponds to a time derivative
+   */
+  virtual bool containsTimeKernel() = 0;
+
+  /**
+   * If the system has a kernel that corresponds to a time derivative
```

Comment:
This doxygen string could be a little better. The current one makes me expect a simple boolean as the return type